cohere_v3_vector/README.md

```diff
@@ -0,0 +1,73 @@
+## Cohere msmarco-v2/embed-english-v3 vector track
+
+This track benchmarks the dataset from [Cohere/msmarco-v2-embed-english-v3](https://huggingface.co/datasets/Cohere/msmarco-v2-embed-english-v3).
+
+Given the size of this dataset 138.3M documents with 1024 dimension vectors you
+need a cluster with at least 60GB of total RAM available to run performant HNSW queries.
+
+### Generating the document dataset
+
+To rebuild the dataset run the following commands, **warning** this takes at least 2.7TB of free disk space:
+
+```console
+$ cd cohere_v3_vector
+$ python3 -m venv .venv
+$ source .venv/bin/activate
+$ pip install -r _tools/requirements.txt
+$ python _tools/parse_documents.py
+
+# Create a test file for each page of documents
+$ for file in cohere-documents-*; do
+  head -n 1000 $file > "${file%.*}-1k.json"
+done
+
+# Zip each document file for uploading
+$ for file in cohere-documents-*; do
+  pv $file | bzip2 -k >> $file.bz2
+done
+$ ls -1 cohere-documents-* > files.txt
+```
+
+This will build 47 `cohere-documents-XX.json` file for the entire dataset of 138.3M documents and then bzip then. Note that this script depends on the libraries listed `_tools/requirements.txt` to run and it takes a few hours to download and parse all the documents.
+### Example Document
+
+```json
+{
+  "docid": "00_0",
+  "title": "0-60 Times - 0-60 | 0 to 60 Times & 1/4 Mile Times | Zero to 60 Car Reviews",
+  "text": "0-60 Times - 0-60 | 0 to 60 Times & 1/4 Mile Times | Zero to 60 Car Reviews.",
+  "emb": [0.00507354736328125, 0.01519775390625, -0.002109527587890625, ...]
+}
+```
+
+### Generating the queries
+
+The `queries.json` can be rebuilt using the `_tools/generate_queries.py`, this will stream a partial training dataset from hugging face, call the Cohere generate API for each to get a question based on it, and then call the Cohere embed API for the question, and store the embeddings in `queries.json`.
+This will take a very long time, maybe grab a ☕️ ?
+
```

Comment:
That's ok if we "just" want to test the performance but ideally we should use the [official queries](https://ir-datasets.com/msmarco-passage-v2.html#msmarco-passage-v2/train). Then in a follow up it will be easy to add similarity measures in the track based on the associated `qrels`.